### PR TITLE
create better-enums package

### PR DESCRIPTION
**Pull request by leodido**
As per title.

### CODE REVIEW

File: packages/b/better-enums/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    set_homepage("http://aantron.github.io/better-enums")
+    set_description("C++ compile-time enum to string, iteration, in a single header file")
+    set_license("BSD-2-Clause")
+
+    add_urls("https://github.com/aantron/better-enums/archive/refs/tags/$(version).tar.gz",
+             "https://github.com/aantron/better-enums.git")
+    add_versions("0.11.3", "1b1597f0aa5452b971a94ab13d8de3b59cce17d9c43c8081aa62f42b3376df96")
+
+    on_install(function (package)
+        print(package:installdir("include"))
```
Comment (waruqi) [inline packages/b/better-enums/xmake.lua:12]: please remove it

File: packages/b/better-enums/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    on_test(function (package)
+        assert(package:check_cxxsnippets({test = [[
+        #include <iostream>
+
+        BETTER_ENUM(Word, int, Hello, World)
+
+        void test() {
+            std::cout << (+Word::Hello)._to_string() << ", " << (+Word::World)._to_string() << "!" << std::endl;
+        }
```
Comment (waruqi) [inline packages/b/better-enums/xmake.lua:25]: please add indent for test code body

File: packages/b/better-enums/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        constexpr const char *identifier_2 = std::enum_traits<Channel>::enumerators::get<2>::identifier;
+
+        static_assert(size == 3, "");
+        static_assert(value_0 == +Channel::Red, "");
+        static_assert(value_1 == +Channel::Green, "");
+
+        void test()
+        {
+            std::cout << identifier_2 << std::endl;
+        }
```
Comment (waruqi) [inline packages/b/better-enums/xmake.lua:51]: indent

File: packages/b/better-enums/xmake.lua (hunk 1 above)
Comment (leodido) [reply packages/b/better-enums/xmake.lua:12]: About to push, done

**Pull request merged**